### ISSUE TITLE
Bootbox closes too agressively

### ISSUE THREAD

**Issue by av01d** [makeusabrew/bootbox#705]
I have the following contents in a bootbox (5.1.0) dialog:

<img width="502" alt="Screenshot 2019-05-13 at 14 11 53" src="https://user-images.githubusercontent.com/6247646/57620456-1a140600-7589-11e9-934d-d3af23df1ce9.png">

The footer (`.modal-footer`) contains a dropdown, a bootstrap `button-group`. Whenever you try to interact with this dropdown, the dialog closes (the dropdown options are never shown).
The cause is this:

```
dialog.on('click', '.modal-footer button:not(.disabled)', function (e) {
      var callbackKey = $(this).data('bb-handler');
      
      processCallback(e, dialog, callbacks[callbackKey]);
});
```

All button elements cause a callback to be called. A small code change fixes the issue:

```
dialog.on('click', '.modal-footer button:not(.disabled)', function (e) {
   var callbackKey = $(this).data('bb-handler');
   if (callbackKey !== undefined) {
      // Only process callbacks for buttons we recognize:
      processCallback(e, dialog, callbacks[callbackKey]);
   }
});
```

**Comment by tiesont** [comment]
Sounds like a good candidate for a pull-request. Otherwise, I agree with what you're showing, but it'll be a bit before it gets integrated.

**Comment by tiesont** [comment]
Implemented in v5.2.0
